src/redux/slices/catalogueSlice.ts

```diff
@@ -138,6 +143,35 @@ const catalogueSlice = createSlice({
         },
       };
     },
+    loadingDeleteProductFromCatalogue: (state: CatalogueState) => {
+      return {
+        ...state,
+        deleteProductFromCatalogue: { loading: true, success: false, error: false },
+      };
+    },
+    successDeleteProductFromCatalogue: (state: CatalogueState, action) => {
+      // const index = state.products.indexOf(action.payload);
+      // if (index > -1) state.products.splice(index, 1);
+      return {
+        ...state,
+        products: state.products.filter((product) => product.id !== action?.payload?.id),
+        deleteProductFromCatalogue: {
+          loading: false,
+          success: true,
+          error: false,
+        },
+      };
+    },
```

Comment:
Ya encontré la respuesta para esto!
[Acá esta](https://stackoverflow.com/questions/34582678/is-this-the-correct-way-to-delete-an-item-using-redux), es la primera respuesta de esta página.
Necesitas el `index` de la línea 153, pero puedes borrar lo de la línea 154.
Después, la línea 157 reemplazala por: `products: [...state.products.slice(0, index), ...state.products.slice(index + 1)],`

Reply:
Diego, con esto una vez que elimino un producto, no se me va lo que tenía antes. Por ejemplo, me muestra la lista [P1, P2, P3] y si elimino el P1 me queda [P1, P2, P3, P2, P3], sabes como arreglar eso?

Reply:
Que raro, veamoslo en discord mejor.
<img width="572" alt="Screen Shot 2021-05-29 at 10 01 09" src="https://user-images.githubusercontent.com/42161460/120073061-ccca3a80-c064-11eb-8526-422b30e0bff6.png">